staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -300,21 +300,20 @@ func (s *store) GuaranteedUpdate(
 		}
 
 		ret, ttl, err := s.updateState(origState, tryUpdate)
-		if err != nil {
-			// It's possible we were working with stale data
-			if mustCheckData && apierrors.IsConflict(err) {
-				// Actually fetch
-				origState, err = getCurrentState()
-				if err != nil {
-					return err
-				}
-				mustCheckData = false
-				// Retry
-				continue
-			}
-
+		if err != nil && !mustCheckData {
```

Comment:
slight preference for the previous structure, since it makes it really clear we either continue or return if err is non-nil. If you want to return early and unnest one level, that's ok as well:

```go
if err != nil {
  // If our data is already up to date, return the error
  if !mustCheckData {
    return err
  }

  // It's possible we were working with stale data
  // Actually fetch
  origState, err = getCurrentState()
  if err != nil {
    return err
  }
  mustCheckData = false
  // Retry
  continue
}
```

Reply:
Applied. Thanks for the suggestion.